### DISCUSSION TITLE
ReadTimeout in asyncio has no message

### DISCUSSION THREAD

**Discussion by PrasannaVenkadesh** [encode/httpx#2681] [Potential Issue]
httpx version: 0.23.3

When a timeout occurs in the sync client, I could see `ReadTimeout: timed out` message while the message is missing in the async client. It's `ReadTimeout: `

**Comment by lovelydinosaur** [comment]
Thanks for noting this.
I've nudged it over into a discussion as "potential issue", which is our starting point for conversations like this.

If anyone would like this escalated into an issue, then...

* Provide a nice simple replicable case.
* Show exactly the behaviour with each of: sync, asyncio, trio.

Thanks. 👍

**Reply by lovelydinosaur** [reply]
(For extra points, show the behaviour directly in `httpcore` instead of using `httpx`.)

**Reply by lahwaacz** [reply]
The output in https://github.com/encode/httpx/discussions/2425 shows that `PoolTimeout` in async context has the same issue.

Here are some more reproducers for `ConnectTimeout`, `ReadTimeout` and potentially also `ReadError`:
```python
import asyncio
import httpx

timeout = httpx.Timeout(1, read=0.1)
client = httpx.AsyncClient(timeout=timeout)

async def main():
    await client.get("http://5digits.org/pentadactyl/")   # raises httpx.ConnectTimeout with empty message
#    await client.get("http://dev.osdrawer.net/")   # raises either httpx.ReadTimeout or httpx.ReadError with empty message
#    await client.get("https://httpbin.org/delay/10")   # raises httpx.ReadTimeout with empty message

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

**Reply by preethiabk** [reply]
We've been seeing the same while using async client and  when there is a delay from the server's response. We get a httpx.ReadTimeout with empty message.

**Reply by karpetrosyan** [reply]
I don't think it's an issue. We are wrapping the lower-level exceptions from libraries like AnyIO, Trio, and the built-in socket, so it should be expected that depending on what you are using (sync/asyncio/Trio), you will get different exceptions for network operations.

Here we are seeing the empty message because anyio.fail_after raises a TimeoutError exception without a message. Maybe we should ask them to add a message for a better user experience?

**Comment by escalate** [comment]
Hello everyone,

As a workaround, I output the name of the exception in addition to the exception message. 

```
try:
    await client.get(url)
except HTTPError as e:
    logger.error(f"{type(e).__name__} {e}")
```

This at least gives me a short but informative error message.

I hope this helps you.

Best regards
Felix

**Reply by lovelydinosaur** [reply]
Hi Felix, thanks.

Would you be able to provide before/after examples in your comment here?

**Reply by escalate** [reply]
You can find the full code here: https://github.com/escalate/aem-dispatcher-security-scan/blob/master/src/aem_dispatcher_security_scan/aem_dispatcher_security_scan.py
